public/app/features/alerting/unified/components/settings/AlertmanagerConfig.tsx

```diff
@@ -0,0 +1,185 @@
+  };
+
+  const {
+    register,
+    setValue,
+    setError,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormValues>({
+    defaultValues,
+  });
+
+  useEffect(() => {
+    if (savingError) {
+      setError('configJSON', { type: 'deps', message: savingError.message });
+    }
+  }, [savingError, setError]);
+
+  useEffect(() => {
+    if (deletingError) {
+      setError('configJSON', { type: 'deps', message: deletingError.message });
+    }
+  }, [deletingError, setError]);
+
+  // manually register the config field with validation
+  register('configJSON', {
+    required: { value: true, message: 'Configuration cannot be empty' },
+    validate: (value: string) => {
+      try {
+        JSON.parse(value);
+        return true;
+      } catch (e) {
+        return e instanceof Error ? e.message : 'JSON is invalid';
+      }
+    },
+  });
+
+  const handleSave = handleSubmit(
+    (values: FormValues) => {
+      onSave(alertmanagerName, defaultValues.configJSON, values.configJSON);
+    },
+    (errors) => {
+      console.error(errors);
+    }
+  );
+
+  const isOperating = isLoadingConfig || isDeleting || isSaving;
+
+  /* loading error, if this fails don't bother rendering the form */
+  if (loadingError) {
+    return (
+      <Alert severity="error" title="Failed to load Alertmanager configuration">
+        {loadingError.message ?? 'An unkown error occurred.'}
+      </Alert>
+    );
+  }
+
+  /* resetting configuration state */
+  if (isDeleting) {
+    return (
+      <Alert severity="info" title="Resetting Alertmanager configuration">
+        It might take a while...
+      </Alert>
+    );
+  }
+
+  return (
+    <div className={styles.container}>
+      {/* form error state */}
+      {errors.configJSON && (
+        <Alert severity="error" title="Oops, something went wrong">
+          {errors.configJSON.message || 'An unknown error occurred.'}
+        </Alert>
+      )}
+
+      {isLoadingSuccessful && (
+        <div className={styles.content}>
+          <AutoSizer>
+            {({ height, width }) => (
+              <>
+                <CodeEditor
+                  language="json"
+                  width={width}
+                  height={height}
+                  showLineNumbers={true}
+                  monacoOptions={{
+                    scrollBeyondLastLine: false,
+                  }}
+                  value={defaultValues.configJSON}
+                  showMiniMap={false}
+                  onSave={(value) => setValue('configJSON', value)}
+                  onBlur={(value) => setValue('configJSON', value)}
+                  readOnly={isOperating}
+                />
+              </>
+            )}
+          </AutoSizer>
+        </div>
+      )}
+
+      {!readOnly && (
+        <Stack justifyContent="flex-end">
+          <Button variant="destructive" disabled={isOperating} onClick={() => setShowResetConfirmation(true)}>
```

Comment:
super nit: the `disabled`/`onClick` props are the other way round in this button compared to the rest 🫠

Reply:
I wonder if there is a prettier rule for consistent prop ordering 🤔